extensions/ql-vscode/src/query-history.ts

```diff
@@ -1267,8 +1267,22 @@ export class QueryHistoryManager extends DisposableObject {
     }
   }
 
-  async handleExportResults(): Promise<void> {
-    await commands.executeCommand('codeQL.exportVariantAnalysisResults');
+  async handleExportResults(
+    singleItem: QueryHistoryInfo,
+    multiSelect: QueryHistoryInfo[],
+  ): Promise<void> {
+    const { finalSingleItem, finalMultiSelect } = this.determineSelection(singleItem, multiSelect);
+
+    if (!this.assertSingleQuery(finalMultiSelect) || !finalSingleItem) {
+      return;
+    }
+
+    // Remote queries and variant analysis only
+    if (finalSingleItem.t === 'remote') {
+      await commands.executeCommand('codeQL.exportRemoteQueryResults', finalSingleItem.queryId);
+    } else if (finalSingleItem.t === 'variant-analysis') {
+      await commands.executeCommand('codeQL.exportVariantAnalysisResults', finalSingleItem.variantAnalysis.id);
+    }
```

Comment:
Is there room for a quick unit test [like we have](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/src/vscode-tests/no-workspace/query-history.test.ts#L402-L410) for `handleCancel`?

Reply:
Thanks for the pointer! I've added unit tests for both this method and the `handleCopyRepoList` which does basically the same thing.

Reply:
This is excellent! 🙌  Thanks @koesie10 !